hack/cert-generator.sh

```diff
@@ -0,0 +1,149 @@
+#!/bin/bash
+
+set -e
+
+service="katib-controller"
+webhook="katib.kubeflow.org"
+secret="katib-webhook-cert"
+namespace="kubeflow"
+fullServiceDomain="${service}.${namespace}.svc"
+
+# Fully qualified name of the CSR object.
+csr="certificatesigningrequests.v1beta1.certificates.k8s.io"
+
+if [ ! -x "$(command -v openssl)" ]; then
+  echo "ERROR: openssl not found"
+  exit 1
+fi
+
+tmpdir=$(mktemp -d)
+echo "INFO: Creating certs in tmpdir ${tmpdir} "
+
+cat <<EOF >>"${tmpdir}/csr.conf"
+[req]
+req_extensions = v3_req
+distinguished_name = req_distinguished_name
+[req_distinguished_name]
+[ v3_req ]
+basicConstraints = CA:FALSE
+keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+extendedKeyUsage = serverAuth
+subjectAltName = @alt_names
+[alt_names]
+DNS.1 = ${service}
+DNS.2 = ${service}.${namespace}
+DNS.3 = ${fullServiceDomain}
+EOF
+
+openssl genrsa -out "${tmpdir}/server-key.pem" 2048
+openssl req -new -key "${tmpdir}/server-key.pem" -subj "/CN=${fullServiceDomain}" -out "${tmpdir}/server.csr" -config "${tmpdir}/csr.conf"
```

Comment:
```suggestion
openssl req -new -key "${tmpdir}/server-key.pem" -subj "/CN=system:node:${fullServiceDomain}/O=system:nodes" -out "${tmpdir}/server.csr" -config "${tmpdir}/csr.conf"
```

Reply:
Make sense, thank you for investigation @tenzen-y!